apps/dashboard/src/components/workflow-editor/steps/configure-step-template-form.tsx

```diff
@@ -27,55 +28,65 @@ export type StepEditorProps = {
   step: StepDataDto;
 };
 
-export type ConfigureStepTemplateFormProps = StepEditorProps & {
-  debouncedUpdate: (data: UpdateWorkflowDto) => void;
+type ConfigureStepTemplateFormProps = StepEditorProps & {
+  update: (data: UpdateWorkflowDto) => void;
 };
-export const ConfigureStepTemplateForm = (props: ConfigureStepTemplateFormProps) => {
-  const { workflow, step, debouncedUpdate } = props;
 
+export const ConfigureStepTemplateForm = (props: ConfigureStepTemplateFormProps) => {
+  const { workflow, step, update } = props;
   const schema = useMemo(() => buildDynamicZodSchema(step.controls.dataSchema ?? {}), [step.controls.dataSchema]);
-
   const defaultValues = useMemo(
     () => merge(buildDefaultValues(step.controls.uiSchema ?? {}), step.controls.values),
-    []
+    [step]
   );
 
   const form = useForm({
-    mode: 'onChange',
     resolver: zodResolver(schema),
     defaultValues: defaultValues,
-    shouldFocusError: true,
```

Comment:
There is an issue with automatically focusing the field that we will have to solve. 
Basically when this option is set to `true` the field receives the focus. And if not fix the issue in the field it's not possible to blur into any another field. That's because `onBlur` implementation triggers the validation again.